### ISSUE TITLE
The build shoul fail on Javadoc linter warnings

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#10815]
```bash
$ cat TestClass.java
```

```java
public class TestClass {
    /**
     * Some desc.
     *
     * @return some value
     * @throws Exception
     * @return some other value
     * 
     */
    public int getValue() throws Exception {}
}
```

```bash
$ javadoc TestClass.java
Loading source file TestClass.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 16.0.1+9
Building tree for all the packages and classes...
Generating ./TestClass.html...
TestClass.java:14: warning: no description for @throws
     * @throws Exception
       ^
TestClass.java:15: warning: @return has already been specified
     * @return some other value
       ^
```

These violations are warnings. The build passes if something similar is found. We should configure the build to fail on such violations.

_Originally posted by @rnveach in https://github.com/checkstyle/checkstyle/pull/10813#discussion_r714377756_
